### PR TITLE
Refactored DateExtensions

### PR DESCRIPTION
**Pull request by EdiLT**
### PR's key points

Renamed and optimized DateExtensions
 
### How to review this PR?
…

### CODE REVIEW

**Comment by sebastianvarela** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          6 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><strong>📢 [iOS]</strong> Executed 138 tests, with 0 failures (0 unexpected) in 4.094 (4.416) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><strong>📢 [mac]</strong> Executed 85 tests, with 0 failures (0 unexpected) in 1.119 (1.281) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><strong>📢 [tv]</strong> Executed 138 tests, with 0 failures (0 unexpected) in 4.230 (4.415) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">⌛️ <strong>[iOS]</strong> Slowest test: test_show_snackbar() took 2.018 seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">⌛️ <strong>[mac]</strong> Slowest test: test_html_value() took 0.681 seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">⌛️ <strong>[tv]</strong> Slowest test: test_show_snackbar() took 2.030 seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Source/Foundation/Extensions/DateExtensions.swift [hunk 1]
```diff
@@ -89,10 +86,50 @@ public extension Date {
...
+
+    /**
+     Calculate months between two dates, included days in last month or not
+     
+     - parameters:
+     - endDate: date to
+     - withDays: boolean that indicate if calculate with days (By default included it)
+     - returns: Amount of months between this dates
+     */
+    func monthsBetween(endDate: Date, withDays: Bool = true) -> Int? {
```
Comment (sebastianvarela) [inline Source/Foundation/Extensions/DateExtensions.swift]: the behavior of this func is not clear. 
`func monthsBetween(_ date: Date, ignoringDays: Bool = false) -> Int`

- Why the response is optional? When the response is nil?
- ignoringDays (with inverse logic) ir more semantic, I think

File: Source/Foundation/Extensions/DateExtensions.swift [hunk 2]
```diff
@@ -35,48 +35,45 @@ public extension Date {
...
 
-    var shortYear: String {
-        return formatWith(dateFormat: "yy")
+    var formattedRFC822Date: String {
+        return DateFormatter.rfc822DateFormatter.string(from: self)
     }
 
-    func formattedSpanishDate(timeZone: TimeZone? = nil) -> String {
-        return formatWith(dateFormat: "dd 'de' MMMM", timeZone: timeZone)
+    var formattedSpanishDay: String {
```
Comment (sebastianvarela) [inline Source/Foundation/Extensions/DateExtensions.swift]:
```suggestion
    var formattedDay: String {
```
Spanish is not relevant here

File: Source/Foundation/Extensions/DateExtensions.swift [hunk 3]
```diff
@@ -35,48 +35,45 @@ public extension Date {
...
+        return format(dateFormat: "d",
+                      formatter: .spanishDateFormatter)
```
Comment (sebastianvarela) [inline Source/Foundation/Extensions/DateExtensions.swift]: you can use the default formatter for this case.

File: Source/Foundation/Extensions/DateExtensions.swift [hunk 4]
```diff
@@ -89,10 +85,50 @@ public extension Date {
         return calendar.isDateInToday(self)
     }
 
+    func formattedSpanishDate(timeZone: TimeZone? = nil) -> String {
+        return format(dateFormat: "dd 'de' MMMM",
+                      timeZone: timeZone,
+                      formatter: .spanishDateFormatter)
+    }
+
+    func formattedSpanishFullDate(timeZone: TimeZone? = nil) -> String {
```
Comment (minuscorp) [inline Source/Foundation/Extensions/DateExtensions.swift]: `func fullDateFormatted(formatter: CountryDateFormatter = .spain, timezone: TimeZone? = nil) -> String

**Comment by EdiLT** [comment]
This PR is included in https://github.com/bq/swift-magic-pills/pull/41

**Pull request closed without merging**